{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Using the unknown bacterium DNA for SPAdes analysis \n",
    "1. locate the location of the files (- cd /data/lab8)  \n",
    "folder contains  \n",
    "1.illumina_reads_R1.fastq – first paired-end read  \n",
    "2.illumina_reads_R2.fastq – second paired-end read  \n",
    "3.pacbio_reads.fastq – long PacBio reads\n",
    "\n",
    "SPAdes is a hybrid genome assembler, meaning that it takes multiple sources of information as\n",
    "input and combines them to produce an optimal assembly.Assemblies using only short reads\n",
    "tend to be highly fragmented (i.e., many contigs). Assemblies using a high-quality short read set and a higher error rate long-read set (like PacBio) tend to be the best.\n",
    "\n",
    "2. using the SPAdes command line as shown below:  \n",
    "***spades -t 1 -1 /data/lab8/illumina_reads_R1.fastq -2 /data/lab8/illumina_reads_R2.fastq --pacbio /data/lab8/pacbio_reads.fastq  -o /home/be131-03/snow/lab8***\n",
    "\n",
    "annotation of the code above:\n",
    "specify the output directory with --o, the path to the first Illumina read with -1, the path to the second Illumina read with -2, and the path to your PacBio reads with --pacbio.\n",
    "\n",
    "3. using the command control+A+D, the screen will continue running even logged off\n",
    "\n",
    "##### Analysis:\n",
    "We expect short reads to produce a more fragmented assembly than long reads, since the the reads are small and arbitrary, on the opposite, the long reads are much more condensed and coherent less likely to be broken up assembly.\n",
    "\n",
    "A single-molecule sequencing like PacBio have a higher error rate than Illumina. Because PacBio read the long-read sets which may be less thorough with sequencing. Illumina reasd the smaller sequences and create multiply contiguous sequences.However, for creating scaffolds, having more coverage is preferable to higher accuracy.\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Generate assembly statistics\n",
    "After the SPAdes runs, there are two files. \"Contigs.fasta\" are contiguous sequences that could be assembled from your reads. \"Scaffolds.fasta\" are sets of contigs that have been stitched together in order, and are generally longer than contigs.\n",
    "\n",
    "Using the assembly-stats program to calculate statistics on both your contigs and\n",
    "scaffolds file.\n",
    "##### Result:\n",
    "stats for contigs.fasta  \n",
    "sum = 6678635, n = 176, ave = 37946.79, largest = 414008  \n",
    "N50 = 234690, n = 11  \n",
    "N60 = 202142, n = 14  \n",
    "N70 = 154232, n = 18  \n",
    "N80 = 103367, n = 23  \n",
    "N90 = 81286, n = 30  \n",
    "N100 = 128, n = 176  \n",
    "N_count = 0  \n",
    "Gaps = 0\n",
    "\n",
    "***total length of all contigs is 6678635 bp, the number of contigs (or scaffolds) is 176, and the N50 is 234690, n=11.***\n",
    "\n",
    "stats for scaffolds.fasta  \n",
    "sum = 6678655, n = 174, ave = 38383.07, largest = 414008  \n",
    "N50 = 234690, n = 11  \n",
    "N60 = 202142, n = 14  \n",
    "N70 = 154232, n = 18  \n",
    "N80 = 103367, n = 23  \n",
    "N90 = 81821, n = 30  \n",
    "N100 = 128, n = 174  \n",
    "N_count = 20  \n",
    "Gaps = 2  \n",
    "***the total length of all scaffolds is 6678655 bp, the number of scaffolds is 174, and the N50 is 234690, n=11.***\n",
    "\n",
    "#### Analysis:\n",
    "\n",
    "the N50 is defined as the sequence length of the shortest contig at 50% of the total genome length.It gives more possible combinations in the total genome pool of contigs. It also gives higer weights to the longer contig but at the same time not ignore the shorter chains \n",
    "\n",
    "\n",
    "The scaffolds approximately equals to the contigs. the value is making sense since there are only two gaps in the scaffolds \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Extra Credit: \n",
    "The number of times a contig in your assembly was covered by the reads used to\n",
    "assemble it (“coverage”) is listed at the end of the contig name in contigs.fasta. Extract the\n",
    "coverage from each FASTA header and plot a histogram of coverage for all contigs in your\n",
    "assembly."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "import matplotlib.pyplot as plt\n",
    "coverages=[]\n",
    "for record in SeqIO.parse(\"contigs.fasta\", \"fasta\"):\n",
    "    coverages.append(float(record.id.split('_')[5]))\n",
    "#print(coverages)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAYIAAAEKCAYAAAAfGVI8AAAABHNCSVQICAgIfAhkiAAAAAlwSFlzAAALEgAACxIB0t1+/AAAADl0RVh0U29mdHdhcmUAbWF0cGxvdGxpYiB2ZXJzaW9uIDMuMC4wLCBodHRwOi8vbWF0cGxvdGxpYi5vcmcvqOYd8AAAEXFJREFUeJzt3X2wXHV9x/H3hwBVojzfoQjaUKQ4qVWwKSJY2wFLqVpBihRGaFQ66PhQrJ0p6HRGa4tFRkWqVScVMTJWnhWqjkoBrQ8UTSAIJFoogsIEEgVE6PgQ+PaPPbdcY5K7wZzdvfm9XzN37u7ZPXc/OZN7P3t+Z8/vpKqQJLVrm3EHkCSNl0UgSY2zCCSpcRaBJDXOIpCkxlkEktQ4i0CSGmcRSFLjLAJJaty24w4wjN13370WLFgw7hiSNKcsX778B1U1Ndvz5kQRLFiwgGXLlo07hiTNKUnuHOZ5Dg1JUuMsAklqnEUgSY2zCCSpcRaBJDXOIpCkxlkEktQ4i0CSGmcRSFLj5sSZxb+KBad/doPL7zjzxSNOIkmTyT0CSWqcRSBJjbMIJKlxFoEkNc4ikKTGWQSS1DiLQJIaZxFIUuMsAklqnEUgSY2zCCSpcRaBJDXOIpCkxlkEktQ4i0CSGmcRSFLjLAJJapxFIEmNswgkqXEWgSQ1ziKQpMZZBJLUOItAkhpnEUhS43otgiR/neSWJDcn+WSSJyTZJ8l1SW5LcmGS7fvMIEnatN6KIMlewF8Bi6rqmcA84HjgXcDZVfV04H7g5L4ySJJm1/fQ0LbAE5NsC+wArAYOAy7pHl8KHN1zBknSJvRWBFV1N/Bu4HsMCuBHwHLggapa1z3tLmCvvjJIkmbX59DQLsBRwD7AU4D5wJGbsf4pSZYlWbZ27dqeUkqS+hwaeiHw3apaW1U/By4DDgV27oaKAPYG7t7QylW1pKoWVdWiqampHmNKUtv6LILvAQcn2SFJgMOBlcA1wLHdcxYDl/eYQZI0iz6PEVzH4KDw9cBN3WstAU4D3pzkNmA34Ny+MkiSZrft7E95/KrqbcDb1lt8O3BQn68rSRqeZxZLUuMsAklqnEUgSY2zCCSpcRaBJDXOIpCkxlkEktQ4i0CSGmcRSFLjLAJJapxFIEmNswgkqXEWgSQ1ziKQpMZZBJLUOItAkhpnEUhS4ywCSWqcRSBJjbMIJKlxFoEkNc4ikKTGWQSS1DiLQJIaZxFIUuMsAklqnEUgSY2zCCSpcRaBJDXOIpCkxlkEktQ4i0CSGmcRSFLjLAJJapxFIEmNswgkqXG9FkGSnZNckuTbSVYleV6SXZNcmeTW7vsufWaQJG1a33sE5wCfr6pnAM8GVgGnA1dV1X7AVd19SdKY9FYESXYCXgCcC1BVP6uqB4CjgKXd05YCR/eVQZI0uz73CPYB1gLnJbkhyUeSzAf2qKrV3XPuAfbY0MpJTkmyLMmytWvX9hhTktrWZxFsCzwH+FBVHQg8zHrDQFVVQG1o5apaUlWLqmrR1NRUjzElqW19FsFdwF1VdV13/xIGxXBvkj0Buu9reswgSZpFb0VQVfcA30+yf7focGAlcAWwuFu2GLi8rwySpNlt2/PPfyPwiSTbA7cDr2JQPhclORm4Eziu5wySpE3otQiqagWwaAMPHd7n60qShueZxZLUOItAkhpnEUhS4ywCSWqcRSBJjRuqCJKcmmTHDJyb5PokR/QdTpLUv2H3CF5dVQ8CRwC7ACcBZ/aWSpI0MsMWQbrvLwLOr6pbZiyTJM1hwxbB8iRfZFAEX0jyZODR/mJJkkZl2DOLTwYOAG6vqv9NshuD6SIkSXPcsHsEV1bV9d2FZaiqHwJn9xdLkjQqm9wjSPIEYAdg9+7awtPHBXYE9uo5myRpBGYbGnoN8CbgKcByHiuCB4EP9JhLkjQimyyCqjoHOCfJG6vq/SPKJEkaoaEOFlfV+5McAiyYuU5VfbynXJKkERmqCJKcD+wLrAAe6RYXYBFI0hw37MdHFwELu4vNS5K2IsN+fPRm4Nf7DCJJGo9h9wh2B1Ym+Qbw0+mFVfXSXlJJkkZm2CJ4e58hJEnjM+ynhr7cdxBJ0ngM+6mhHzP4lBDA9sB2wMNVtWNfwSRJozHsHsGTp28nCXAUcHBfoSRJo7PZl6qsgU8Df9xDHknSiA07NHTMjLvbMDiv4Ce9JJIkjdSwnxr60xm31wF3MBgekiTNccMeI/AiNJK0lRrqGEGSvZN8Ksma7uvSJHv3HU6S1L9hDxafB1zB4LoETwH+vVsmSZrjhi2Cqao6r6rWdV8fA6Z6zCVJGpFhi+CHSU5MMq/7OhH4YZ/BJEmjMWwRvBo4DrgHWA0cC7yyp0ySpBEa9uOj7wAWV9X9AEl2Bd7NoCAkSXPYsHsEz5ouAYCqug84sJ9IkqRRGrYItkmyy/Sdbo9g2L0JSdIEG/aP+XuAa5Nc3N1/OXDGMCsmmQcsA+6uqpck2Qe4ANgNWA6cVFU/27zYkqQtZag9gqr6OHAMcG/3dUxVnT/ka5wKrJpx/13A2VX1dOB+4OTh40qStrShZx+tqpVV9YHua+Uw63RnH78Y+Eh3P8BhwCXdU5YCR29eZEnSlrTZ01BvpvcBfws82t3fDXigqtZ19+8C9uo5gyRpE3orgiQvAdZU1fLHuf4pSZYlWbZ27dotnE6SNK3PPYJDgZcmuYPBweHDgHOAnZNMH6TeG7h7QytX1ZKqWlRVi6amnM1CkvrSWxFU1Vuqau+qWgAcD1xdVa8ArmFwZjLAYuDyvjJIkmbX9zGCDTkNeHOS2xgcMzh3DBkkSZ2RnBRWVV8CvtTdvh04aBSvK0ma3Tj2CCRJE8QikKTGWQSS1DiLQJIaZxFIUuMsAklqnEUgSY2zCCSpcRaBJDXOIpCkxlkEktQ4i0CSGmcRSFLjLAJJapxFIEmNswgkqXEWgSQ1ziKQpMZZBJLUOItAkhpnEUhS4ywCSWqcRSBJjbMIJKlxFoEkNc4ikKTGWQSS1DiLQJIaZxFIUuMsAklqnEUgSY2zCCSpcRaBJDXOIpCkxlkEktQ4i0CSGtdbESR5apJrkqxMckuSU7vluya5Msmt3fdd+sogSZpdn3sE64C/qaqFwMHA65MsBE4Hrqqq/YCruvuSpDHprQiqanVVXd/d/jGwCtgLOApY2j1tKXB0XxkkSbMbyTGCJAuAA4HrgD2qanX30D3AHhtZ55Qky5IsW7t27ShiSlKTei+CJE8CLgXeVFUPznysqgqoDa1XVUuqalFVLZqamuo7piQ1q9ciSLIdgxL4RFVd1i2+N8me3eN7Amv6zCBJ2rQ+PzUU4FxgVVW9d8ZDVwCLu9uLgcv7yiBJmt22Pf7sQ4GTgJuSrOiWvRU4E7goycnAncBxPWaQJM2ityKoqq8C2cjDh/f1ui1ZcPpnN7j8jjNfPOIkkuYyzyyWpMZZBJLUOItAkhpnEUhS4ywCSWqcRSBJjbMIJKlxFoEkNc4ikKTGWQSS1Lg+5xrSmDj1hKTN4R6BJDXOIpCkxlkEktQ4jxE0xGMHkjbEPQJJapxFIEmNswgkqXEWgSQ1ziKQpMZZBJLUOD8+Ogds7GOfkrQluEcgSY2zCCSpcRaBJDXOIpCkxlkEktQ4i0CSGmcRSFLjLAJJapwnlGkir1MwiZmkrZV7BJLUOPcI1rOl3ok+nmkhJu3dru/KpTa4RyBJjXOPQE1yb0d6zFj2CJIcmeQ7SW5Lcvo4MkiSBkZeBEnmAf8C/AmwEDghycJR55AkDYxjaOgg4Laquh0gyQXAUcDKUYbY3IO5o7gmwFy/7sAohlv6fg2HjB7jthidcW/rcQwN7QV8f8b9u7plkqQxSFWN9gWTY4Ejq+ovu/snAc+tqjes97xTgFO6u/sD3xnyJXYHfrCF4vbJnFvOXMgIcyPnXMgI5hzWb1TV1GxPGsfQ0N3AU2fc37tb9guqagmwZHN/eJJlVbXo8ccbDXNuOXMhI8yNnHMhI5hzSxvH0NA3gf2S7JNke+B44Iox5JAkMYY9gqpal+QNwBeAecBHq+qWUeeQJA2M5YSyqvoc8LmefvxmDyeNiTm3nLmQEeZGzrmQEcy5RY38YLEkabI415AkNW6rKoK5MnVFkjuS3JRkRZJl484DkOSjSdYkuXnGsl2TXJnk1u77LuPM2GXaUM63J7m7254rkrxozBmfmuSaJCuT3JLk1G75RG3PTeSctO35hCTfSHJjl/Pvu+X7JLmu+32/sPvwyaRl/FiS787YlgeMK+OmbDVDQ93UFf8N/BGDk9S+CZxQVSM9Y3kYSe4AFlXVxHwOOskLgIeAj1fVM7tlZwH3VdWZXbHuUlWnTWDOtwMPVdW7x5ltWpI9gT2r6vokTwaWA0cDr2SCtucmch7HZG3PAPOr6qEk2wFfBU4F3gxcVlUXJPkwcGNVfWjCMr4W+ExVXTKOXMPamvYI/n/qiqr6GTA9dYWGUFX/Cdy33uKjgKXd7aUM/kiM1UZyTpSqWl1V13e3fwysYnD2/ERtz03knCg18FB3d7vuq4DDgOk/sGPdnpvIOCdsTUUwl6auKOCLSZZ3Z1BPqj2qanV3+x5gj3GGmcUbknyrGzoa+xDWtCQLgAOB65jg7bleTpiw7ZlkXpIVwBrgSuB/gAeqal33lLH/vq+fsaqmt+UZ3bY8O8mvjTHiRm1NRTCXPL+qnsNgBtbXd8MdE60GY4iT+g7nQ8C+wAHAauA9440zkORJwKXAm6rqwZmPTdL23EDOidueVfVIVR3AYCaCg4BnjDnSL1k/Y5JnAm9hkPX3gF2BsQ6tbszWVARDTV0xCarq7u77GuBTDP5jT6J7u3Hk6fHkNWPOs0FVdW/3S/go8K9MwPbsxokvBT5RVZd1iydue24o5yRuz2lV9QBwDfA8YOck0+dCTczv+4yMR3bDb1VVPwXOY4K25UxbUxHMiakrkszvDsyRZD5wBHDzptcamyuAxd3txcDlY8yyUdN/XDsvY8zbsztweC6wqqreO+OhidqeG8s5gdtzKsnO3e0nMvhAyCoGf2yP7Z421u25kYzfnlH8YXAMYyJ/17eaTw0BdB9zex+PTV1xxpgj/ZIkv8lgLwAGZ3b/2yTkTPJJ4A8ZzJZ4L/A24NPARcDTgDuB46pqrAdqN5LzDxkMYxRwB/CaGWPxI5fk+cBXgJuAR7vFb2Uw/j4x23MTOU9gsrbnsxgcDJ7H4M3rRVX1ju536QIGQy43ACd277wnKePVwBQQYAXw2hkHlSfGVlUEkqTNtzUNDUmSHgeLQJIaZxFIUuMsAklqnEUgSY2zCCSpcRaBtAXNONNVmjMsAjUjyV90k3/dmOT8JAuSXN0tuyrJ05LslOTOJNt068xP8v0k2yXZN8nnu8kCv5LkGd1zPpbkw0muA85KclCSa5PckOTrSfbvnrdDkosymP//U91c+ou6x47o1rk+ycXd/D/SSPjuRU1I8tvA3wGHVNUPkuzK4EzQpVW1NMmrgX+uqqO7GST/gMEUBi8BvlBVP0+yhMGZobcmeS7wQQZTIcNgrptDquqRJDsCv19V65K8EHgn8GfA64D7q2phNyHZii7b7l22F1bVw0lOYzDX/jtGsW0ki0CtOAy4ePpiQFV1X5LnAcd0j58PnNXdvhD4cwZFcDzwwe4d+iHAxYNpYwCYOaXwxVX1SHd7J2Bpkv0YTNOwXbf8+cA53evfnORb3fKDgYXA17qfvT1w7Zb4R0vDsAikX3YF8M5ur+F3gauB+Qzmv9/YpQYfnnH7H4Brqupl3Tz/X5rl9cJg/voTfpXQ0uPlMQK14mrg5Ul2g8H1g4GvM3jHD/AKBhOw0U0K9k0G794/003J/CDw3SQv79ZPkmdv5LV24rEpkV85Y/nXGFwGkiQLgd/plv8XcGiSp3ePzU/yW7/aP1cankWgJlTVLcAZwJeT3Ai8F3gj8KpuiOYkBteYnXYhcGL3fdorgJO79W9h45dCPQv4pyQ38It73R8EppKsBP6x+xk/qqq1DArjk12Wa5nAC69o6+Xso9KIJJkHbFdVP0myL/AfwP7dNbalsfEYgTQ6OwDXdFcFC/A6S0CTwD0CSWqcxwgkqXEWgSQ1ziKQpMZZBJLUOItAkhpnEUhS4/4P73YXoNMZXEAAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 432x288 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.hist(coverages[0:len(coverages)-2],50)\n",
    "plt.ylabel('counts')\n",
    "plt.xlabel('coverage')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Analysis:\n",
    "    The result is bimodal distribution ( extreme value distribution). Most coverages occure within 10. The mean coverage is about 1.5. But there are few exceptional points exist at 25 and 32. \n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Identify the taxon from which your genome originated\n",
    "Much of the gene is highly conserved, as function ribosomes are required for protein synthesis, but some regions differ greater between bacterial taxa. These “hypervariable regions” can be used to determine the taxon from which a 16S rRNA gene originated.\n",
    "\n",
    "To identify your genome, you must   \n",
    "##### 1) scan over the entire genome to locate copies of the 16S ribosomal RNA gene\n",
    "\n",
    "run the rna_hmm3.py program on your assembled contigs to locate rRNA genes. Must\n",
    "specify a path to the input assembly with -i and a path to the output General Feature Format\n",
    "(GFF) annotation file with -o.\n",
    "After this program completes, manually delete all lines within other than those containing 16S_rRNA genes.\n",
    "\n",
    "The code used in the terminal is     \n",
    "\n",
    "*rna_hmm3.py -i contigs.fasta -o contigs.gff*\n",
    "\n",
    "##### 2) extract the 16S rRNA genes from your assembly  \n",
    "use bedtools getfasta to extract nucleic acid sequences of the 16S rRNA gene.Specify the path to contigs.fasta with -fi and the path to the GFF file you obtained above, with -bed.\n",
    "\n",
    "The code used in the terminal is    \n",
    "\n",
    "*getfasta -fi contigs.fasta -bed output_16S_rRNA.gff -fo rRNAgene.fasta*\n",
    "\n",
    "##### 3) compare these genes to a database of known 16S rRNA genes.  \n",
    "the resulting FASTA file from the previous step uploads into the Ribosomal Database Project’s SeqMatch tool to identify the 16S sequences as precisely as possible by comparing them to a database from known bacteria."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Result:\n",
    "NODE_23_length_103367_cov_10.081286:35530-37075\n",
    "rootrank Root (1) (query sequences)\n",
    "\n",
    "    domain Bacteria  (1)   \n",
    "\n",
    "        phylum \"Proteobacteria\"  (1)  \n",
    "\n",
    "           class Gammaproteobacteria  (1)      \n",
    "\n",
    "             order Pseudomonadales  (1)     \n",
    "\n",
    "                family Pseudomonadaceae  (1)      \n",
    "\n",
    "                   genus Pseudomonas  (1)      \n",
    "NODE_18_length_154232_cov_9.584186:124541-126086\n",
    "rootrank Root (1) (query sequences)\n",
    "\n",
    "    domain Bacteria  (1)      \n",
    "\n",
    "        phylum \"Proteobacteria\"  (1)     \n",
    "\n",
    "          class Gammaproteobacteria  (1)       \n",
    "\n",
    "             order Pseudomonadales  (1)       \n",
    "\n",
    "                family Pseudomonadaceae  (1)       \n",
    "\n",
    "                 genus Pseudomonas  (1)      \n",
    "NODE_5_length_327223_cov_9.241629:46285-47830\n",
    "rootrank Root (1) (query sequences)\n",
    "\n",
    "    domain Bacteria  (1)      \n",
    "\n",
    "        phylum \"Proteobacteria\"  (1)      \n",
    "\n",
    "           class Gammaproteobacteria  (1)     \n",
    "\n",
    "             order Pseudomonadales  (1)      \n",
    "\n",
    "                 family Pseudomonadaceae  (1)      \n",
    "\n",
    "                     genus Pseudomonas  (1)       \n",
    "NODE_6_length_298784_cov_8.918606:13925-15470\n",
    "rootrank Root (1) (query sequences)\n",
    "\n",
    "    domain Bacteria  (1)      \n",
    "\n",
    "        phylum \"Proteobacteria\"  (1)      \n",
    "\n",
    "          class Gammaproteobacteria  (1)      \n",
    "\n",
    "             order Pseudomonadales  (1)      \n",
    "\n",
    "                family Pseudomonadaceae  (1)      \n",
    "\n",
    "                   genus Pseudomonas  (1)        \n",
    "NODE_4_length_332509_cov_9.114046:137011-138556\n",
    "rootrank Root (1) (query sequences)\n",
    "\n",
    "    domain Bacteria  (1)       \n",
    "\n",
    "       phylum \"Proteobacteria\"  (1)      \n",
    "\n",
    "          class Gammaproteobacteria  (1)       \n",
    "\n",
    "             order Pseudomonadales  (1)       \n",
    "\n",
    "                family Pseudomonadaceae  (1)      \n",
    "\n",
    "                   genus Pseudomonas  (1)        \n",
    "NODE_2_length_378948_cov_10.006665:348499-350044\n",
    "rootrank Root (1) (query sequences)\n",
    "\n",
    "    domain Bacteria  (1)      \n",
    "   \n",
    "        phylum \"Proteobacteria\"  (1)      \n",
    "\n",
    "          class Gammaproteobacteria  (1)      \n",
    "\n",
    "             order Pseudomonadales  (1)       \n",
    "\n",
    "                family Pseudomonadaceae  (1)      \n",
    "\n",
    "                   genus Pseudomonas  (1) \n",
    "\n",
    "#### Analysis: \n",
    "According to the results above, we used 6 16S rRNA to test. All of them are coming from the  Pseudomonas genus level. For the species level,it is hard to determine which exact species it comes from based on the just one rRNA sequence. The result listed 20 possible species. "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Genome annotation\n",
    "look inside the genome and determine with high confidence whether it contains genes necessary to produce a cell wall, or virulence factors enabling the infection of an animal host. \n",
    "\n",
    "1. Using Genome annotation,it breaks the long genome sequence into genes, then identify their function by identifying orthologs with known function in other, more well-studied genomes\n",
    "\n",
    "- two remote annotation services that will perform the annotation automatically: RAST and BASys.\n",
    "\n",
    "#### Result:\n",
    "![Rast result](Rast.png)\n",
    "![BASys result](BASys.png)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Research and write-up\n",
    "\n",
    "Using the Ribosomal Database Project’s SeqMatch tool, we had identified that the unknown bacteria potentially comes from the genus level of Pseudomonas. Pseudomonas is a gram-negative stained bacteria. From the RAST result, the unknown bacteria shows that it has the total length of 6,678,635 bp, a GC content of 58.7%, and N50 of 234690 and L50 of 11. The total number of the contigs with PEGs is 176. There are 393 subsystems, 6352 coding sequences and 85 RNAs.\n",
    "\n",
    "The subsystem coverage in overall is 29%. Saying the feature that are encoded by the genome, the largest percentage is the amino acid and its derivatives, containing the polyamines metabolism, urea cycle and arginine biosynthesis extended. The second largest group is the carbohydrates, including multiple carbohydrate metabolism pathways, fermentation and one-carbon metabolism. and the third largest is the cofactors, vitamins, prosthetic groups, and pigments, including biotin and thiamin biosynthesis, lipoic acid metabolism and so on. The tested bacteria has no pathways encoded for the photosynthesis. \n",
    "\n",
    "Members of the genus Pseudomonas inhabit a wide variety of environments, which is reflected in their versatile metabolic capacity and broad potential for adaptation to fluctuating environmental conditions[1]. Horizontal gene transfer has impacted the capability of pathogenic Pseudomonas spp. in terms of disease severity and specificity. For example, Genes most likely to be required for nitrogen fixation and rhizosphere competence were identified in a 49-kb region containing 59 genes in Psedomonas. stutzeri A1501, which, based on the G-C content, was postulated to be a GI acquired through horizontal transfer, and inserted between cobS (PST_1301) and PST_1360. \n",
    "\n",
    "The pseudomonas genome has high probability of encoding antibiotics resistance genes that inhibit the antibiotics activation. One study has determined that Pseudomonas aeruginosa can live in a biofilm mode within hypoxic mucus in the airways of patients with cystic fibrosis (CF) [2]. The biofilm leads to the result of multi- antibiotic resistance.  The mechanism behind this event is that the coded protein enzyme β-lactamase or penicillinase can hydrolyze the β-lactam ring of the antibiotics and penicillin, as the result, the antibiotics and penicillin become inactivation. In the paper, the research introduces the idea of using a metallo-complex, desferrioxamine-gallium (DFO-Ga) that targets P. aeruginosa iron metabolism. The use of DFO-Ga as a horse delivery system that interferes with iron metabolism shows promise as a treatment for P. aeruginosa infections.\n",
    "\n",
    "\n",
    "Reference:  \n",
    "[1] Silby, Mark W., et al. \"Pseudomonas genomes: diverse and adaptable.\" FEMS microbiology reviews 35.4 (2011): 652-680.  \n",
    "\n",
    "[2] Banin, Ehud, et al. \"The potential of desferrioxamine-gallium as an anti-Pseudomonas therapeutic agent.\" Proceedings of the National Academy of Sciences 105.43 (2008): 16761-16766.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
